v2/middlewares/sigint/sigint.go

```diff
@@ -0,0 +1,30 @@
+package sigint
+
+import (
+	"context"
+	"os"
+	"os/signal"
+
+	"github.com/honestbank/kp/v2/middlewares"
+)
+
+type sigInt[T any] struct{}
+
+func (s sigInt[T]) Process(ctx context.Context, item T, next func(ctx context.Context, item T) error) error {
+	return next(ctx, item)
+}
+
+func registerInterruptSignal(cb func()) {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
```

Comment:
Could you make this customizable? we need `syscall.SIGTERM, syscall.SIGINT, syscall.SIGQUIT, syscall.SIGHUP`

Reply:
are you sure? why do we need all that?

Reply:
k8s sends a SIGINT when a pod needs to stop, correct?

Reply:
will do.